<?xml version="1.0"?>

<launch>
    <arg name="node_name" default="mppi_node_throttle"/>
    <!-- File management -->
    <arg name="learning_config_file" default="../../../../configs/yamaha_atv/atv_model_0.1.yaml"/>
    <arg name="use_gridmaps" default="true"/>
    <arg name="kbm_config" default="$(find torch_mpc)/configs/kbm_20.yaml"/>
    <arg name="model_fp" default=""/>

    <!-- Common MPPI parameters -->
    <arg name="mppi_iterations" default="1"/>
    <arg name="mppi_throttle_noise" default="0.1"/>
    <arg name="mppi_steer_noise" default="0.1"/>
    <arg name="mppi_temperature" default="0.02"/>
    <arg name="mppi_use_ou" default="true"/>
    <arg name="mppi_use_all_last_controls" default="false,false"/>
    <arg name="mppi_use_normal_init_noise" default="true,true"/>
    <arg name="mppi_ou_alpha" default="[0.0, 0.0]"/>
    <arg name="mppi_ou_scale" default="[3.25, 1.25]"/>
    <arg name="mppi_d_ou_scale" default="[1.0, 3.0]"/>
    <arg name="mppi_H" default="20"/>
    <arg name="mppi_sH" default="0"/>
    <arg name="mppi_throttle_delay" default="0.5"/>
    <!--  -->

    <!-- Model specific MPPI parameters -->
    <arg name="pitch_thresh" default="0.1"/>
    <arg name="pitch_window" default="10"/>
    <arg name="kbm_model_name" default="gravity"/>
    <arg name="rgb_list" default="1,1,1"/>
    <arg name="mppi_N" default="2048"/>
    <arg name="mppi_uN" default="1024"/>
    <arg name="device" default="cpu"/>
    <!--  -->

    <rosparam command="load" file="$(find torch_mpc)/configs/cost_params.yaml"/>

	<node pkg="torch_mpc" type="$(arg node_name).py" name="$(arg node_name)" output="screen">
        <param name="config_spec_path" value="$(dirname)/$(arg learning_config_file)"/>
        <param name="costmap_topic" value="/local_costmap"/>
        <param name="gridmap_topic" value="/local_gridmap"/>
        <param name="waypoint_feature_topic" value="/next_waypoints/odom"/>
        <param name="odom_topic" value="/integrated_to_init"/>
        <param name="base_frame" value="sensor_init"/>
        <param name="use_gridmaps" value="$(arg use_gridmaps)"/>

        <param name="mppi_iterations" value="$(arg mppi_iterations)"/>
        <param name="mppi_throttle_noise" value="$(arg mppi_throttle_noise)"/>
        <param name="mppi_steer_noise" value="$(arg mppi_steer_noise)"/>
        <param name="mppi_temperature" value="$(arg mppi_temperature)"/>
        <param name="mppi_use_ou" value="$(arg mppi_use_ou)"/>
        <param name="mppi_use_all_last_controls" value="$(arg mppi_use_all_last_controls)"/>
        <param name="mppi_use_normal_init_noise" value="$(arg mppi_use_normal_init_noise)"/>
        <param name="mppi_ou_alpha" value="$(arg mppi_ou_alpha)"/>
        <param name="mppi_ou_scale" value="$(arg mppi_ou_scale)"/>
        <param name="mppi_d_ou_scale" value="$(arg mppi_d_ou_scale)"/>
        <param name="mppi_N" value="$(arg mppi_N)"/>
        <param name="mppi_uN" value="$(arg mppi_uN)"/>
        <param name="mppi_H" value="$(arg mppi_H)"/>
        <param name="mppi_sH" value="$(arg mppi_sH)"/>
        <param name="mppi_throttle_delay" value="$(arg mppi_throttle_delay)"/>
        <param name="mppi_device" value="$(arg device)"/>
        <param name="kbm_config" value="$(arg kbm_config)"/>
        <param name='rgb_list' value="$(arg rgb_list)"/>
        <param name='pitch_window' value="$(arg pitch_window)"/>
        <param name='pitch_thresh' value="$(arg pitch_thresh)"/>
        <param name="kbm_model_name" value="$(arg kbm_model_name)"/>
        <param name="model_fp" value="$(arg model_fp)"/>
	</node>

</launch>
